src/main/java/seedu/apollo/storage/Storage.java

```diff
@@ -273,19 +274,39 @@ private static ModuleList readModuleFileContents(File save, Ui ui, ModuleList al
                 if (newModule == null) {
                     throw new InvalidSaveFile();
                 }
+
                 Module module = new Module(newModule.getCode(), newModule.getTitle(), newModule.getModuleCredits());
+                if (isAdded(newModuleList, module)) {
+                    throw new DuplicateModuleInTextFileException();
+                }
                 addLessons(module, newModule, moduleInfoArgs);
                 calendar.addModule(module);
                 newModuleList.add(module);
                 counter++;
             } catch (InvalidSaveFile e) {
                 ui.printInvalidSaveFile(counter, filePath);
+            } catch ( DuplicateModuleInTextFileException e) {
+                ui.printDuplicateModuleInTextFile(counter,moduleDataFilePath);
             }
         }
         return newModuleList;
     }
 
-
+    /**
+     * Checks if the module is already in the module file.
+     *
+     * @param moduleList The list of modules.
+     * @param module The module to be checked.
+     * @return True if the module is already in the module file.
+     */
+    public static boolean isAdded(ModuleList moduleList, Module module) {
+        for (Module mod: moduleList) {
+            if (mod.getCode().equals(module.getCode())) {
```

Comment:
I think it would be better to have more descriptive variable names so that the reader knows where mod and module are located.